### ISSUE TITLE
Ensure ddev is properly utilizing Semantic Versioning

### ISSUE THREAD

**Issue by rickmanelius** [drud/ddev#352]
## What happened (or feature request):

* Clarification

## What you expected to happen:

* Ensure our roadmap and release strategy is fully compliant with http://semver.org/.

## Remaining Actions

* [x] Document our commitment in our documentation and community guidelines.

**Comment by beeradb** [comment]
The good news is we actually are fully compliant!



items #4 and #5 from semver.org: 

- Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.
- Version 1.0.0 defines the public API. The way in which the version number is incremented after this release is dependent on this public API and how it changes.

tl;dr: since we're pre-1.0.0, anything is permissible. This has been (at least on my part) purposeful, as needing to track breaking changes as things have been moving so fast hasn't seemed worth the effort. We are approaching a phase of feature 'completion' and relative stability, though.

I think the meat of this ticket is actually determining where our 1.0.0 should be, and then ensuring we're more diligent about being accurate with respect to major/minor/patch releases after that.

**Comment by rickmanelius** [comment]
@beeradb Agreed with everything you said regarding focusing on how this impacts post 1.0.0. One small logistical/admin item is to update the roadmap versions to include both decimal points and probably be a little more careful on what's truly 1.1.0 and 1.2.0 until we've evaluated if/when APIs change.

**Comment by rickmanelius** [comment]
While not completely related to this, there were other point releases that didn't have an official milestone attached. I might want to back create it so that we have better linking to issues and when they were released along the roadmap.

**Comment by rickmanelius** [comment]
So having thought through this it a bit more and going back to source reference material, I think this impacts us in the following ways.

* Technically, we've been adopting a "v" to prepend the version number to provide a visual queue to an end-user that this is a version number and not something else. Techncially this isn't true semver and it's assumed that "v1.2.3" would be an acceptable tag with the assumed semver number of "1.2.3". See https://github.com/semver/semver/blob/master/semver.md#is-v123-a-semantic-version as a reference.
* Branches. I envision a time when we want to get a major item reviewed and committed while also not being ready to officially incremement to a new major release. I'm therefore assuming that we will want to adapt our workflow to include a second branch one major release ahead for these situations, although that might become untenable depending on how much drift occurs.
* Reviews. We will probably need to have as part of our code and end-user review process a confirmation that no major backward-incompatible BC's were introduced. Otherwise, this is all for show and we're not going to hold ourselves accountable. Since we already have an official review process for releases, I suggest we add our fail-safe reviews there as the final stop gap, although that might be challenging because it's already committed to master and essentially forcing a major version bump or a revert of the changes. In that case, we may need to require this as part of our code reviews per feature.
* Release Notes. For major releases where incompatible changes and deprecation occur, we need to have a more robust means of communicating those changes in our release notes.

From an issue scope perspective, I think this boils down to:

1. Do we commit to this publicly? Default answer: yes.
2. Do we maintain a branch open for next major release? Default answer: yes.
3. Where do we document? Default answers: our commitment in our community guides and the process in our review requirements.
4. Do we drop the "v" label in our versioning? Default answer: yes unless we feel this adds enough value.

Open to feedback from anyone on the team. I would also like a thumbs up from at least @cyberswat or @beeradb on the technical items and my default answers to the questions above. Once approved I'll open a PR on the suggested doc changes (if applicable) as well as update our milestone labeling moving forward.

**Comment by rfay** [comment]
I don't have a particular attachment to the 'v' prefix, but github seems to, see https://github.com/drud/ddev/releases/new  (instructions). I can't remember or never knew why this is.  But we explicitly switched to the 'v' because if this issue I think.

**Comment by rickmanelius** [comment]
Good to know because I didn't recall that tidbit from our history. FWIW, I prefer maintaining v just because it always prompts me to think "version".

**Comment by rickmanelius** [comment]
Based on a quick internal discussion. It seems like maintaining the "v" is preferred because 1) we've been doing it already 2) other very popular and semver compatible projects like Kube also use it 3) it doesn't hurt us or give us anything truly important by dropping.

**Comment by rickmanelius** [comment]
Updated issue summary to reflect the one remaining action (documentation in our readthedocs and community guidelines).

**Comment by rickmanelius** [comment]
While I'm unwavering on the decision to proceed with Semver support, it's curious to hear dissenting voices. Posting for posterity...

* https://gist.github.com/jashkenas/cbd2b088e20279ae2c8e
* http://akkartik.name/post/versioning

**Comment by rickmanelius** [comment]
PRs merged. Closing!

**Comment by rfay** [comment]
No comments on why this might have been reopened, so I'm closing it.

**Comment by rickmanelius** [comment]
Re-opened so we can update our documentation :)

**Comment by rickmanelius** [comment]
As per an internal conversation, the gains of fully committing to semver is not readily apparent and is creating some artificial dependencies. To that tend, we will revert the PRs in our community docs and DDEV documentation that explicitly state support. Then we can consider this closed.

**Comment by rfay** [comment]
As in slack, I don't think we've finished this conversation, and I don't even think it's a valid conversation for ddev (semver does not even apply to a tool like semver, IMO). ddev is not a library that you build against.  I think we're free to pursue any line of releases that tries to keep our customers happy. Baffled how we're still not communicating about this after a year of conversations.

**Comment by rickmanelius** [comment]
We'll consider this closed.
